Modules/Scripted/SegmentStatistics/SegmentStatisticsCalculators/ScalarVolumeSegmentStatisticsCalculator.py

```diff
@@ -0,0 +1,167 @@
+      grayscaleNode.GetParentTransformNode(), segmentationToReferenceGeometryTransform)
+
+    cubicMMPerVoxel = reduce(lambda x,y: x*y, referenceGeometry_Reference.GetSpacing())
+    ccPerCubicMM = 0.001
+
+    segment = segmentationNode.GetSegmentation().GetSegment(segmentID)
+    segBinaryLabelName = vtkSegmentationCore.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName()
+    segmentLabelmap = segment.GetRepresentation(segBinaryLabelName)
+
+    segmentLabelmap_Reference = vtkSegmentationCore.vtkOrientedImageData()
+    vtkSegmentationCore.vtkOrientedImageDataResample.ResampleOrientedImageToReferenceOrientedImage(
+      segmentLabelmap, referenceGeometry_Reference, segmentLabelmap_Reference,
+      False, # nearest neighbor interpolation
+      False, # no padding
+      segmentationToReferenceGeometryTransform)
+
+    # We need to know exactly the value of the segment voxels, apply threshold to make force the selected label value
+    labelValue = 1
+    backgroundValue = 0
+    thresh = vtk.vtkImageThreshold()
+    thresh.SetInputData(segmentLabelmap_Reference)
+    thresh.ThresholdByLower(0)
+    thresh.SetInValue(backgroundValue)
+    thresh.SetOutValue(labelValue)
+    thresh.SetOutputScalarType(vtk.VTK_UNSIGNED_CHAR)
+    thresh.Update()
+
+    #  Use binary labelmap as a stencil
+    stencil = vtk.vtkImageToImageStencil()
+    stencil.SetInputData(thresh.GetOutput())
+    stencil.ThresholdByUpper(labelValue)
+    stencil.Update()
+
+    stat = vtk.vtkImageAccumulate()
+    stat.SetInputData(grayscaleNode.GetImageData())
+    stat.SetStencilData(stencil.GetOutput())
+    stat.Update()
+
+    # create statistics list
+    stats = {}
+    if "Scalar Volume.voxel_count" in requestedKeys:
+      stats["Scalar Volume.voxel_count"] = stat.GetVoxelCount()
+    if "Scalar Volume.volume_mm3" in requestedKeys:
+      stats["Scalar Volume.volume_mm3"] = stat.GetVoxelCount() * cubicMMPerVoxel
+    if "Scalar Volume.volume_cc" in requestedKeys:
+      stats["Scalar Volume.volume_cc"] = stat.GetVoxelCount() * cubicMMPerVoxel * ccPerCubicMM
+    if stat.GetVoxelCount()>0:
+      if "Scalar Volume.min" in requestedKeys:
+        stats["Scalar Volume.min"] = stat.GetMin()[0]
+      if "Scalar Volume.max" in requestedKeys:
+        stats["Scalar Volume.max"] = stat.GetMax()[0]
+      if "Scalar Volume.mean" in requestedKeys:
+        stats["Scalar Volume.mean"] = stat.GetMean()[0]
+      if "Scalar Volume.stdev" in requestedKeys:
+        stats["Scalar Volume.stdev"] = stat.GetStandardDeviation()[0]
+    return stats
+
+  def getMeasurementInfo(self, key):
+
+    scalarVolumeNode = slicer.mrmlScene.GetNodeByID(self.getParameterNode().GetParameter("ScalarVolume"))
+
+    try:
+      scalarVolumeQuantity = scalarVolumeNode.GetVoxelValueQuantity()
+      scalarVolumeUnits = scalarVolumeNode.GetVoxelValueUnits()
+    except AttributeError:
```

Comment:
@che85 I think I understand the reason for your change. Two lines above `scalarVolumeNode` is set to
`slicer.mrmlScene.GetNodeByID(self.getParameterNode().GetParameter("ScalarVolume"))`
which might return none `None` or a node not having the method `GetVoxelValueUnits()`
I'll add a comment instead of removing the try/except block